From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Praveen James <pjames@nvidia.com>
Date: Sat, 29 Apr 2023 03:52:44 +0000
Subject: [PATCH 061/241] NVIDIA: SAUCE: tegra-epl: fix deinit notification to
 fsi driver

fix for sc7 failure with FSI disabled build

Bug 4082254

Change-Id: I99149967cb6d9094fbfdee43e97b9042db6b211d
Signed-off-by: Praveen James <pjames@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-jammy/+/2919185
Reviewed-on: https://git-master.nvidia.com/r/c/linux-stable/+/2965784
Tested-by: Prathamesh Shete <pshete@nvidia.com>
Reviewed-by: Prathamesh Shete <pshete@nvidia.com>
Reviewed-by: Laxman Dewangan <ldewangan@nvidia.com>

diff --git a/drivers/platform/tegra/tegra-epl.c b/drivers/platform/tegra/tegra-epl.c
index d7b7cae727906..da33dd765179d 100644
--- a/drivers/platform/tegra/tegra-epl.c
+++ b/drivers/platform/tegra/tegra-epl.c
@@ -78,6 +78,8 @@ static DEFINE_MUTEX(hs_state_mutex);
 
 static struct task_struct *fsi_handshake_thread;
 
+static bool enable_deinit_notify;
+
 static void tegra_hsp_tx_empty_notify(struct mbox_client *cl,
 					 void *data, int empty_value)
 {
@@ -265,10 +267,11 @@ static int epl_client_fsi_handshake(void *arg)
 
 static int __maybe_unused epl_client_suspend(struct device *dev)
 {
-	int ret;
+	int ret = 0;
 	pr_debug("tegra-epl: suspend called\n");
 
-	ret = epl_client_fsi_pm_notify(PM_SUSPEND);
+	if (enable_deinit_notify)
+		ret = epl_client_fsi_pm_notify(PM_SUSPEND);
 	mutex_lock(&hs_state_mutex);
 	hs_state = HANDSHAKE_PENDING;
 	mutex_unlock(&hs_state_mutex);
@@ -381,6 +384,9 @@ static int epl_client_probe(struct platform_device *pdev)
 		}
 	}
 
+	if (of_property_read_bool(np, "enable-deinit-notify"))
+		enable_deinit_notify = true;
+
 	mission_err_status_va = devm_platform_ioremap_resource(pdev, NUM_SW_GENERIC_ERR * 2);
 	if (IS_ERR(mission_err_status_va)) {
 		isAddrMappOk = false;
@@ -402,8 +408,9 @@ static void epl_client_shutdown(struct platform_device *pdev)
 {
 	pr_debug("tegra-epl: shutdown called\n");
 
-	if (epl_client_fsi_pm_notify(PM_SHUTDOWN) < 0)
-		pr_err("Unable to send notification to fsi\n");
+	if (enable_deinit_notify)
+		if (epl_client_fsi_pm_notify(PM_SHUTDOWN) < 0)
+			pr_err("Unable to send notification to fsi\n");
 
 	mutex_lock(&hs_state_mutex);
 	hs_state = HANDSHAKE_PENDING;
-- 
2.43.0

